pkg/modulewriter/hcl_utils.go

```diff
@@ -17,11 +17,24 @@ package modulewriter
 import (
 	"fmt"
 	"path/filepath"
+	"regexp"
 
 	"github.com/hashicorp/hcl/v2/hclwrite"
 	"github.com/zclconf/go-cty/cty"
 )
 
+func escapeBlueprintVariables(hclBytes []byte) []byte {
+	// Convert \$(not.variable) to $(not.variable)
+	re := regexp.MustCompile(`\\\\\$\((.*?)\)`)
```

Comment:
A couple suggestions here:
1. You could drop the final `\)` and the captured text, since it wouldn't matter in the final expression. Example below of what I'm thinking.
1. You should also check that the backslash itself isn't being escaped. If it is, both should be kept.

Example 1:
```go
re := regexp.MustCompile(`\\\\\$\(`)
return re.ReplaceAll(hclBytes, []byte(`$(`)
```

Example 2:
```go
re := regexp.MustCompile(`[^\\]\\\\\$\((.*?)\)`)
return re.ReplaceAll(hclBytes, []byte(`$(${1})`))
```

Combined:
```go
re := regexp.MustCompile(`[^\\]\\\\\$\(`)
return re.ReplaceAll(hclBytes, []byte(`$(`)
```

Reply:
1. Yes. Let's make it as simple as possible.
2. If I understood correctly, It actually already happens

Let's say we have a blueprint file:

```yaml
dummyAttributeName = abc \\$(escape.backslash) abc
```

golang add 2 escape backslash `\` and when converted to bytes it is doubled:

```golang
dummyAttributeName = \"abc \\\\\\\\$(escape.backslash) abc\"\n\n"
```

Calling func `escapeBlueprintVariables()` removes 4 backslashs:

```golang
dummyAttributeName = \"abc \\\\$(escape.backslash) abc\"\n\n"
```

Then when go writes to file:

```hcl
dummyAttributeName = abc \\$(escape.backslash) abc
```